### ISSUE TITLE
(Multiselect): A button must have "no interactive content descendant"

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#5880]
When a user selects one or more items in our multiselect component they're presented with a button inside of a button in order to clear their selection:
![image](https://user-images.githubusercontent.com/40970507/79508228-bc8d0280-7ffe-11ea-9752-bb5d932579f8.png)
![Annotation 2020-04-16 163952](https://user-images.githubusercontent.com/40970507/79509338-f0692780-8000-11ea-99a4-619a75e58879.png)

Along with creating a host of accessibility errors/adding to potential screen reader and keyboard user confusion and frustration this violates [W3C HTML5 spec 4.10.6](https://www.w3.org/TR/html51/sec-forms.html#the-button-element) which states "there must be no interactive content descendant" for a button element.
